storage/innobase/include/fsp_binlog.h

```diff
@@ -464,11 +471,13 @@ extern ulong ibb_page_size;
 extern uint64_t current_binlog_state_interval;
 extern mysql_mutex_t active_binlog_mutex;
 extern pthread_cond_t active_binlog_cond;
+extern mysql_mutex_t binlog_durable_mutex;
+extern mysql_cond_t binlog_durable_cond;
 extern std::atomic<uint64_t> active_binlog_file_no;
 extern uint64_t first_open_binlog_file_no;
 extern uint64_t last_created_binlog_file_no;
-extern std::atomic<uint64_t> binlog_cur_written_offset[2];
-extern std::atomic<uint64_t> binlog_cur_end_offset[2];
+extern std::atomic<uint64_t> binlog_cur_durable_offset[4];
```

Comment:
@knielsen just skimming some of your latest commits, not a formal review, but one thought here would be to have sub-classes for a durable vs non-durable reader, instead of the same reader which is always set up for durable & non-durable reads with a flag in the constructor that tells it which mode to use.